frontend/packages/kubevirt-plugin/integration-tests/tests/vm.cloud-init.scenario.ts

```diff
@@ -0,0 +1,62 @@
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import { withResource } from '@console/shared/src/test-utils/utils';
+import { getCloudInitUserData } from '../../src/selectors/vm/selectors';
+import { basicVMConfig } from './utils/mocks';
+import { CloudInitConfig, ProvisionConfig } from './utils/types';
+import { getResourceObject } from './utils/utils';
+import { vmConfig, CONFIG_NAME_CONTAINER } from './vm.wizard.configs';
+import { VirtualMachine } from './models/virtualMachine';
+
+describe('Kubevirt create VM using cloud-init', () => {
+  const leakedResources = new Set<string>();
+
+  const sourceContainer = 'kubevirt/fedora-cloud-container-disk-demo';
+  const provisionConfig: ProvisionConfig = {
+    provision: {
+      method: CONFIG_NAME_CONTAINER,
+      source: sourceContainer,
+    },
+    networkResources: [],
+    storageResources: [],
+  };
+
+  const hostnameAndKey: CloudInitConfig = {
```

Comment:
there is no useCloudInit anymore - it is resolved just by presence of these values in advanced tab.